feat/src/main/java/com/example/feat/search/SearchScreen.kt

```diff
@@ -0,0 +1,241 @@
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalConfiguration
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.example.core.components.BottomSheet
+import com.example.core.components.RefreshableList
+import com.example.core.components.Skeleton
+import com.example.core.components.VerticalList
+import com.example.core.components.textField.SearchTextField
+import com.example.core.lib.constants.DesignSystem
+import com.example.feat.R
+import com.example.feat.search.components.ContentCategoriesBottomSheet
+import com.example.feat.search.components.ItemCourseSearch
+import com.example.feat.search.controller.SearchUiEvent
+import com.example.feat.search.controller.SearchViewModel
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun SearchScreen(searchViewModel: SearchViewModel) {
+    val uiState by searchViewModel.uiState.collectAsState()
+    val configuration = LocalConfiguration.current
+    val screenWidth = configuration.screenWidthDp
+    val searchController = remember { mutableStateOf("") }
+    val scope = rememberCoroutineScope()
+    val isBottomSheetVisible = remember { mutableStateOf(false) }
+    val sheetState = rememberModalBottomSheetState(
+        skipPartiallyExpanded = true
+    )
+
+    Scaffold(
+        modifier = Modifier
+            .fillMaxSize()
+            .background(color = Color.White.copy(alpha = 0.5f)),
+        topBar = {
+            SearchScreenHeader()
+        }
+    ) { innerPadding ->
+        Box(
+            modifier = Modifier
+                .fillMaxSize()
+                .padding(innerPadding)
+        ) {
+            Column {
+                SearchBar(
+                    searchController = searchController,
+                    scope = scope,
+                    isBottomSheetVisible = isBottomSheetVisible,
+                    sheetState = sheetState,
+                    filterApplied = uiState.selectedContentCategory != null,
+                ) {
+                    searchViewModel.onEvent(SearchUiEvent.OnSearchSubmitted(it))
+                }
+                if (uiState.isLoading)
+                    VerticalList(
+                        size = 5,
+                        item = { _ ->
+                            Skeleton(
+                                width = screenWidth.toDouble(),
+                                height = 150.0,
+                            )
+                        },
+                    )
+                if (uiState.courses.isNotEmpty())
+                    RefreshableList(
+                        size = uiState.courses.size,
+                        isLoading = uiState.isLoading,
```

Comment:
Using this code will cause the refresh component to display when the initial data or search is present.